### PR TITLE
Fix maximum depth bug

### PR DESCRIPTION
**Pull request by keithradford**
Fixed a bug where the maximum depth was being exceeded from the `useEffect` call in `TermButtons.tsx`. The variable `terms` was being called within the method which caused an infinite loop.

### CODE REVIEW

File: src/app/header/components/TermButtons.tsx [hunk 1]
```diff
@@ -4,20 +4,20 @@ import React, { useContext, useEffect, useState } from 'react';
...
 import { getCurrentTerms, getReadableTerm } from '../../shared/utils/terms';
 
+const terms = getCurrentTerms();
+
 export function TermButtons(): JSX.Element {
   const [status, setStatus] = useState([false, false, false]);
 
   const { term, setTerm } = useContext(TermContext);
 
-  const terms = getCurrentTerms();
```
Comment (aomi) [inline src/app/header/components/TermButtons.tsx:12]: Should this be a React hook (or wrapped into a React hook)?

Comment (keithradford) [reply src/app/header/components/TermButtons.tsx:12]: Not sure if it really needs to be. To me this is just a global variable being used for reference in the component.

Comment (aomi) [reply src/app/header/components/TermButtons.tsx:12]: If you're 100% sure we won't have issues where this being used in different places (and not being in sync) will cause any issues then I think we're fine to leave it as is.

**Review by aomi — APPROVED** [review]
If you're sure we won't have issues where `getCurrentTerms` being used in different places (and not being in sync) will not cause any issues then I think we're fine to leave it as is.

File: src/app/header/components/TermButtons.tsx [hunk 2]
```diff
@@ -4,20 +4,20 @@ import React, { useContext, useEffect, useState } from 'react';
 import { TermContext } from '../../context/TermContext';
 import { getCurrentTerms, getReadableTerm } from '../../shared/utils/terms';
 
+const terms = getCurrentTerms();
```
Comment (isaaccormack) [inline src/app/header/components/TermButtons.tsx:7]: Why is this moved to the global scope? Shouldn't just removing `terms` from the deps fix the issue?

Comment (keithradford) [reply src/app/header/components/TermButtons.tsx:7]: either works, is there a preferred way?

Comment (isaaccormack) [reply src/app/header/components/TermButtons.tsx:7]: Well if it doesn't have to be global it probably shouldn't be

Comment (keithradford) [reply src/app/header/components/TermButtons.tsx:7]: I'll just throw the argument out there for it being global: there's a linting rule which wants terms in the dependency array when it's scoped to the function so technically the dependency array will be incomplete. AFAIK both ways fix the bug but both also have "cons".

Comment (isaaccormack) [reply src/app/header/components/TermButtons.tsx:7]: ahh ok thanks, global then

**Review by isaaccormack — APPROVED** [review]
